### ISSUE TITLE
Uncaught channel closed

### ISSUE THREAD

**Issue by howardluong** [platformio/platformio-atom-ide-terminal#556]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.28.0 x64
**Electron**: 2.0.3
**OS**: Microsoft Windows 10 Home
**Thrown From**: [platformio-ide-terminal](https://github.com/platformio/platformio-atom-ide-terminal) package 2.8.3

### Stack Trace

Uncaught channel closed

```
At events.js:183

Error [ERR_IPC_CHANNEL_CLOSED]: channel closed
    at ChildProcess.target.send (internal/child_process.js:588:16)
    at Task.module.exports.Task.send (~/AppData/Local/atom/app-1.28.0/resources/app/src/task.js:113:33)
    at PlatformIOTerminalView.module.exports.PlatformIOTerminalView.resize (/packages/platformio-ide-terminal/lib/view.coffee:276:17)
    at PlatformIOTerminalView.module.exports.PlatformIOTerminalView.resizeTerminalToView (/packages/platformio-ide-terminal/lib/view.coffee:503:6)
    at PlatformIOTerminalView.module.exports.PlatformIOTerminalView.focus (/packages/platformio-ide-terminal/lib/view.coffee:465:6)
    at PlatformIOTerminalView.focus (/packages/platformio-ide-terminal/lib/view.coffee:1:1)
    at /packages/platformio-ide-terminal/lib/status-bar.coffee:105:25
```

### Commands

```
 11x -1:09.1.0 core:backspace (input.hidden-input)
     -0:36 core:save (input.hidden-input)
 15x -0:33 platformio-ide-terminal:toggle (input.hidden-input)
```

### Non-Core Packages

```
platformio-ide-terminal 2.8.3 
```

**Comment by ThePiyushGupta** [comment]
where to put these commands

**Comment by tsepeto** [comment]
Is there a solution?

**Comment by kptnew3** [comment]
Terminal doesn't open in atom instead it raises the error as listed above every time. Its very annoying. Please help.

**Comment by the-j0k3r** [comment]
Yea, follow https://flight-manual.atom.io/hacking-atom/sections/debugging/#reset-to-factory-defaults

then make sure Atom is version 1.29.0 and reinstall platformio-ide-terminal 2.8.4

**Comment by amarnag93** [comment]
Performed factory reset and reinstalled platformio-ide-terminal but still have the issue

**Comment by the-j0k3r** [comment]
What atom version is this you have?

Please provide exact steps to reproduce.

FYI I cant duplicate this at all.

```
> atom -v

Atom    : 1.29.0
Electron: 2.0.5
Chrome  : 61.0.3163.100
Node    : 8.9.3

> apm -v

apm  1.19.0
npm  3.10.10
node 6.9.5 x64
atom 1.29.0
python 3.5.2
git 2.18.0.windows.1
visual studio
```
OS windows 10 pro 1803
Platformio-ide-terminal: 2.8.4

Steps:

1. Ive opened a file and added some content
1. bring up command palette <kbd>ctrl</kbd>  + <kbd>shift</kbd> + <kbd>p</kbd>
1. enter above commands in that order.

```
core:backspace
core:save
platformio-ide-terminal:toggle
```

Everything runs normally no errors no issues.

So my guess is you and everyone else experiencing  have something OS side thats causing this.

**Comment by amarnag93** [comment]
My atom version is 1.29.0.

What should I check from os side?

**Comment by the-j0k3r** [comment]
No idea. Depends on OS, depends on Software, depends on which other atom extensions you have installed.

Its best to work backwards, if you can setup a Virtual machine with same OS fresh and then add atom, platformio-ide-terminal then test...

Then keep adding Atom extensions one by one testing in between until issue manifests, or adding software in same way one by one until issue manifests, add configurations again same manner.

Once you find one that causes error you known.

**Comment by LeoKarz** [comment]
This happened to me. I uninstalled Origin and the games I had downloaded, and then BOOM. It worked.
Don't ask me how it worked. I don't know. I just know it did.

**Comment by the-j0k3r** [comment]
> Don't ask me how it worked. I don't know. I just know it did.

Im not surprised that it worked, more often than not issues are caused by conflicts that external sources add, when you experience an issue in A after installing B, the cause is B, yet people open reports in A.

**Comment by datafiend** [comment]
Ok, still crashing for me. 
I'm on atom 1.3, win10 x64. I've just started with atom and I did install terminate package as well - and that wasn't working either. 
I did get an error that said, "microsoft visual c++ library...ASSERTION FAILED." I read on another thread, # 332. Basically you need to uninstall platformio. Then install GCC+. then re-install platformio. oy vey!

**Comment by kptnew3** [comment]
I was unable to sort it out so switched to visual studio code..

**Comment by the-j0k3r** [comment]
Should be resolved with platformio-ide-terminal 2.9.0

**Comment by the-j0k3r** [comment]
Also duplicate of #613 @ivankravets pls close.
